src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private double randomNumber;
+
+    public Figure getRandomFigure() {
+        getRandomFigureNumber();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        if (randomNumber == 0) {
+            return new Circle("RandomCircle",
+                    colorSupplier.getRandomColor().toString(),
+                    getRandomSide());
+        } else if (randomNumber == 1) {
+            return new Square("RandomSquare",
+                    colorSupplier.getRandomColor().toString(),
+                    getRandomSide());
+        } else if (randomNumber == 2) {
+            return new Rectangle("RandomRectangle",
+                    getDefaultFigure().getColor().toString(),
+                    getRandomSide(),
+                    getRandomSide());
+        } else if (randomNumber == 3) {
+            return new IsoscelesTrapezoid("RandomIsoscelesTrapezoid",
+                    getDefaultFigure().getColor().toString(),
+                    getRandomSide(),
+                    getRandomSide(),
+                    getRandomSide());
+        } else {
+            return new RightTriangle("RandomRightTriangle",
+                    getDefaultFigure().getColor().toString(),
+                    getRandomSide(),
+                    getRandomSide());
+        }
+    }
+
+    //get random number for chose random what figure create
+    public void getRandomFigureNumber() {
```

Comment:
Methods used only in the class must be private
```suggestion
    private void getRandomFigureNumber() {
```